python/pyarrow/tests/test_compute.py

```diff
@@ -108,3 +111,29 @@ def test_take_dictionary(ordered):
     assert result.to_pylist() == ['a', 'b', 'a']
     assert result.dictionary.to_pylist() == ['a', 'b', 'c']
     assert result.type.ordered is ordered
+
+
+@pytest.mark.parametrize(('ty', 'values'), all_array_types)
+def test_filter(ty, values):
+    arr = pa.array(values, type=ty)
+
+    mask = pa.array([True, False, False, True, None])
+    result = arr.filter(mask)
+    result.validate()
+
+    expected = pa.array([values[0], values[3], None], type=ty)
+    assert result.equals(expected)
+
+
+def test_filter_invalid():
```

Comment:
should this test be parameterized as well?

Reply:
Done